packages/synapse-react-client/src/components/TimelinePlot/TimelinePlotSpeciesSelector.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { act, render, screen } from '@testing-library/react'
+import React from 'react'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import { SynapseClient } from '../../index'
+import queryResultBundleJson from '../../mocks/query/syn51735464GroupBySpecies'
+import TimelinePlotSpeciesSelector, {
+  TimelinePlotSpeciesSelectorProps,
+} from './TimelinePlotSpeciesSelector'
+import userEvent from '@testing-library/user-event'
+import { cloneDeep } from 'lodash-es'
+
+const setSpecies = jest.fn()
+
+const timelineSpeciesSelectorProps: TimelinePlotSpeciesSelectorProps = {
+  sql: 'select * from syn123',
+  setSpecies,
+}
+
+const singleRowQueryResultBundle = cloneDeep(queryResultBundleJson)
+singleRowQueryResultBundle.queryResult!.queryResults = {
+  ...singleRowQueryResultBundle.queryResult!.queryResults,
+  rows: [
+    {
+      values: ['Mus musculus'],
+    },
+  ],
+}
+
+const emptyQueryResultBundle = cloneDeep(queryResultBundleJson)
+emptyQueryResultBundle.queryResult!.queryResults = {
+  ...emptyQueryResultBundle.queryResult!.queryResults,
+  rows: [],
+}
+
+async function renderTimelineSelector(
+  props: TimelinePlotSpeciesSelectorProps = timelineSpeciesSelectorProps,
+) {
+  let component
+  // eslint-disable-next-line @typescript-eslint/require-await
+  await act(async () => {
+    component = render(<TimelinePlotSpeciesSelector {...props} />, {
+      wrapper: createWrapper(),
+    })
+  })
+  return component
+}
+
+describe('TimelinePlotSpeciesSelector tests', () => {
+  jest
+    .spyOn(SynapseClient, 'getFullQueryTableResults')
+    .mockResolvedValueOnce(queryResultBundleJson)
+    .mockResolvedValueOnce(singleRowQueryResultBundle)
+    .mockResolvedValueOnce(emptyQueryResultBundle)
```

Comment:
Personally I would prefer setting the appropriate mock in each test. Implicitly relying on test order kinda scares me

Reply:
I agree, this was based on an example.  I'll updated...